miio/fan_miot.py

```diff
@@ -0,0 +1,285 @@
+        "swing_mode_angle": {"siid": 2, "piid": 5},
+        "power_off_time": {"siid": 2, "piid": 6},
+        "buzzer": {"siid": 2, "piid": 8},
+        "light": {"siid": 2, "piid": 7},
+        "mode": {"siid": 2, "piid": 3},
+        "set_move": {"siid": 2, "piid": 9},
+    },
+}
+
+
+class OperationModeMiot(enum.Enum):
+    Normal = 0
+    Nature = 1
+
+
+class FanStatusMiot:
+    """Container for status reports from the Xiaomi Mi Smart Pedestal Fan DMaker P9/P10."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return "on" if self.data["power"] else "off"
+
+    @property
+    def is_on(self) -> bool:
+        """True if device is currently on."""
+        return self.data["power"]
+
+    @property
+    def mode(self) -> OperationMode:
+        """Operation mode."""
+        return OperationMode[OperationModeMiot(self.data["mode"]).name]
+
+    @property
+    def speed(self) -> int:
+        """Speed of the motor."""
+        return self.data["fan_speed"]
+
+    @property
+    def oscillate(self) -> bool:
+        """True if oscillation is enabled."""
+        return self.data["swing_mode"]
+
+    @property
+    def angle(self) -> int:
+        """Oscillation angle."""
+        return self.data["swing_mode_angle"]
+
+    @property
+    def delay_off_countdown(self) -> int:
+        """Countdown until turning off in seconds."""
+        return self.data["power_off_time"]
+
+    @property
+    def led(self) -> bool:
+        """True if LED is turned on, if available."""
+        return self.data["light"]
+
+    @property
+    def buzzer(self) -> bool:
+        """True if buzzer is turned on."""
+        return self.data["buzzer"]
+
+    @property
+    def child_lock(self) -> bool:
+        """True if child lock is on."""
+        return self.data["child_lock"]
+
+    def __repr__(self) -> str:
+        s = (
+            "<FanStatus power=%s, "
+            "mode=%s, "
+            "speed=%s, "
+            "oscillate=%s, "
+            "angle=%s, "
+            "led=%s, "
+            "buzzer=%s, "
+            "child_lock=%s, "
+            "delay_off_countdown=%s>"
+            % (
+                self.power,
+                self.mode,
+                self.speed,
+                self.oscillate,
+                self.angle,
+                self.led,
+                self.buzzer,
+                self.child_lock,
+                self.delay_off_countdown,
+            )
+        )
+        return s
+
+    def __json__(self):
+        return self.data
+
+
+class FanMiot(MiotDevice):
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+        model: str = MODEL_FAN_P10,
+    ) -> None:
+        super().__init__(MIOT_MAPPING[model], ip, token, start_id, debug, lazy_discover)
```

Comment:
This will cause an exception if the model is not in `MIOT_MAPPING`, so it should be moved after the check below.